--- CMakeLists.txt	2020-02-29 16:04:16.000000000 -0500
+++ CMakeLists.txt	2020-02-29 16:03:36.000000000 -0500
@@ -6,9 +6,17 @@
 if(NOT ANDROID)
   find_package(PythonLibs)
   if(PYTHONLIBS_VERSION_STRING VERSION_LESS 3)
-    find_package(Boost REQUIRED python)
+   if(APPLE)
+      find_package(Boost REQUIRED COMPONENTS filesystem python27)
+    else()
+      find_package(Boost REQUIRED COMPONENTS filesystem python)
+    endif()
   else()
-    find_package(Boost REQUIRED python3)
+    if(APPLE)
+      find_package(Boost REQUIRED COMPONENTS filesystem python)
+    else()
+      find_package(Boost REQUIRED COMPONENTS filesystem python3)
+    endif()
   endif()
 else()
 find_package(Boost REQUIRED)
@@ -45,4 +53,4 @@
 install(
   DIRECTORY include/${PROJECT_NAME}/
   DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
-)
\ No newline at end of file
+)
