--- CMakeLists.txt	2020-06-02 13:56:00.000000000 -0400
+++ CMakeLists.txt	2020-06-02 13:56:53.000000000 -0400
@@ -40,11 +40,6 @@
                                       ${PROJECT_NAME}
 )
 
-# Analyzer loader allows other users to test that Analyzers load
-add_executable(analyzer_loader test/analyzer_loader.cpp
-                               gtest-1.7.0/gtest-all.cc)
-target_link_libraries(analyzer_loader diagnostic_aggregator)
-
 if(CATKIN_ENABLE_TESTING)
   find_package(rostest REQUIRED)
   add_rostest(test/launch/test_agg.launch)
@@ -69,6 +64,3 @@
 install(TARGETS ${PROJECT_NAME}
         DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
 )
-install(TARGETS aggregator_node analyzer_loader
-        DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-)
